spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java

```diff
@@ -153,67 +150,104 @@ public RestTemplate getContainerRestTemplate(boolean skipSslVerification, boolea
+					removeAuthorization(httpHeaders);
+					removeAuthorization(entries);
+					httpClientRequest.headers(httpHeaders);
 				}
-		};
-		SSLContext sslContext = SSLContext.getInstance("SSL");
-		// Install trust manager to SSL Context.
-		sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
-
-		// Create a RestTemplate that uses custom request factory
-		return initRestTemplate(
-				httpClientBuilder(sslContext),
-				withHttpProxy,
-				extra);
+			});
+		}
+		catch (SSLException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private boolean shouldRemoveAuthorization(HttpClientRequest request, Map<String, String> extra) {
+		HttpMethod method = request.method();
+		if(method.equals(HttpMethod.GET) || method.equals(HttpMethod.HEAD)) {
+			if (request.uri().contains(AMZ_CREDENTIAL)) {
+				return true;
+			}
+			if (request.uri().contains(AZURECR_URI_SUFFIX)) {
+				return request.requestHeaders()
+					.entries()
+					.stream()
+					.anyMatch(entry -> entry.getKey().equalsIgnoreCase(AUTHORIZATION_HEADER)
+							&& entry.getValue().contains(BASIC_AUTH));
+			}
+			return extra.containsKey(CUSTOM_REGISTRY) && request.uri().contains(extra.get(CUSTOM_REGISTRY));
+		}
+		return false;
 	}
-	private HttpClientBuilder httpClientBuilder(SSLContext sslContext) {
-		// Register http/s connection factories
-		Lookup<ConnectionSocketFactory> connSocketFactoryLookup = RegistryBuilder.<ConnectionSocketFactory> create()
-			.register("https", new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE))
-			.register("http", new PlainConnectionSocketFactory())
-			.build();
-		return HttpClients.custom()
-			.setConnectionManager(new BasicHttpClientConnectionManager(connSocketFactoryLookup));
+
+	private static void removeAuthorization(HttpHeaders headers) {
+		for(Map.Entry<String,String> entry: headers.entries()) {
+			if(entry.getKey().equalsIgnoreCase(org.springframework.http.HttpHeaders.AUTHORIZATION)) {
+				headers.remove(entry.getKey());
+				break;
+			}
+		}
 	}
-	private RestTemplate initRestTemplate(HttpClientBuilder clientBuilder, boolean withHttpProxy, Map<String, String> extra) {
 
-		clientBuilder.setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(StandardCookieSpec.RELAXED).build());
+	private RestTemplate initRestTemplate(HttpClient client, boolean withHttpProxy, Map<String, String> extra) {
 
 		// Set the HTTP proxy if configured.
 		if (withHttpProxy) {
 			if (!properties.getHttpProxy().isEnabled()) {
 				throw new ContainerRegistryException("Registry Configuration uses a HttpProxy but non is configured!");
 			}
-			HttpHost proxy = new HttpHost(properties.getHttpProxy().getHost(), properties.getHttpProxy().getPort());
-			clientBuilder.setProxy(proxy);
+			ProxyProvider.Builder builder = ProxyProvider.builder().type(ProxyProvider.Proxy.HTTP).host(properties.getHttpProxy().getHost()).port(properties.getHttpProxy().getPort());
+			client.proxy(typeSpec -> builder.build());
 		}
```

Comment:
Breaking the lines on the fluent builder makes this more readable I think. You like?

```suggestion
			ProxyProvider.Builder builder = ProxyProvider.builder().type(ProxyProvider.Proxy.HTTP)
					.host(properties.getHttpProxy().getHost())
					.port(properties.getHttpProxy().getPort());
```

Reply:
done